Introduce --debug flag

Vincinator opened the pull request:
* by default the pkg build container execute a single script, and the container will be removed after execution. The new "--debug" flag makes sure that the container is not removed and the workingdir for creating the sources are not removed on completion or on error

**What this PR does / why we need it**:
* if we want to create patches, we need to prepare the source folder of the package and perform some manual tasks on the sources. See https://github.com/gardenlinux/package-build/pull/4 

Fixes #3 

**Special notes for your reviewer**:
You may want to cherry-pick the commit [10a0c29](https://github.com/gardenlinux/package-build/pull/6/commits/10a0c29cb99c6394c9e481a5b6d6b64107d83d58) into this PR: https://github.com/gardenlinux/package-build/pull/4

Vincinator reviewed (approved):
Looks good. 

Maybe we need to rename the `scripts` in the future. `scripts` does not intuitively tell me that it is used for github actions only. 

But works for me for now

alee-ntap commented:
> Maybe we need to rename the `scripts` in the future. `scripts` does not intuitively tell me that it is used for github actions only.

Sure. Any suggestion for the name?

Vincinator commented:
Hm, maybe it would help if we could move the scripts folder to `.github/scripts/`, then it would be clear that it is for github actions.

alee-ntap commented:
After one more review and discuss with @Vincinator . It makes more sense to move everything will be installed into the container into `container/` folder. And then leave the rest on the root path eg: the `misc/mini_sudo.c` and `scripts/gh_release`. I will update these into this PR.

Vincinator committed:
Introduce --debug flag

* by default the pkg build container execute a single script,
and the container will be removed after execution.
The new "--debug" flag makes sure that the container is not removed
and the workingdir for creating the sources are not removed on completion
or on error

fwilhe commented on bin/source:
```diff
@@ -95,8 +95,15 @@ message="Rebuild for Garden Linux."
+  trap 'cd / && rm -rf "$dir"' EXIT
+else
+  trap 'echo $dir' EXIT
+  echo "Linux sources are located in:"
```
I wrote linux here because at the time we were working on package-linux, but I think in this general context it's confusing that way.

```suggestion
  echo "Package sources are located in:"
```

fwilhe commented on build:
```diff
@@ -63,7 +68,12 @@ if [ -n "$build_dep_dir" ]; then
+  else
+    echo "Starting package build container without --rm"
+	  podman run --security-opt seccomp=unconfined --arch "$arch" "${mount_opts[@]}" -w "/opt/package_build/workdir" -e gl_package_build_debug=true "$container" /opt/package_build/bin/source
+  fi
```
Should the same logic also apply for the binary build? I think chances are higher that there are issues in the source build, but depending on how the package is built issues might also come up in the binary part (I had that in one package so far, not sure if it's common)

Open question, I'm fine the way it is.

fwilhe reviewed (approved):
I have not tried it out, but i looks good to me. Thanks for taking care.

fwilhe committed:
Update bin/source
